aten/src/ATen/WrapDimUtils.h

```diff
@@ -45,6 +45,15 @@ static inline int64_t maybe_wrap_dim(int64_t dim, const std::vector<std::vector<
   return maybe_wrap_dim(dim, tensor_sizes[0].size());
 }
 
+// wrap negative dims in to_transform_dims
+static inline void maybe_wrap_dims(std::vector<int64_t>& to_transform_dims, int64_t tensor_total_dims) {
+  for (size_t i = 0; i < to_transform_dims.size(); i++) {
+    if (to_transform_dims[i] < 0) {
+      to_transform_dims[i] = (tensor_total_dims + (to_transform_dims[i] % tensor_total_dims)) % tensor_total_dims;
```

Comment:
This expression is pretty weird. If you are taking modulo of `tensor_total_dims`, why are you adding it at the beginning?

Reply:
Oh, this is to handle the negative case. But then, what if I have `to_transform_dims[i] < -tensor_total_dims`?

Reply:
I admit that I was lazy and just copied this from @weiyangfb 's #8628 PR without looking much at it..... 

Yeah, I'll comment on that PR.